### PR TITLE
HPCC-16446 Fix various warnings in the code generator

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please check carefully.  They are mainly removing unused code, but check I haven't messed up something.

I have run the regression suite.

File: ecl/hql/hqlgram.y [hunk 1]
```diff
@@ -8933,7 +8933,7 @@ simpleDataSet
                             IHqlExpression * arg = $7.getExpr();
                             if (isStringType(arg->queryType()))
                             {
-                                arg = createAttribute(nameAtom, arg);
+                                arg = createExprAttribute(nameAtom, arg);
```
Comment (richardkchapman) [inline ecl/hql/hqlgram.y]: This seems like a more substantive change. How/why did coverity flag it? What is the effect?

**Comment by richardkchapman** [comment]
@ghalliday Couple of questions

File: ecl/hql/hqlgram.y (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlgram.y]: It spotted that code just below it in the same file was calling a createExprAttribute - and labelled it as a copy/paste error.  The changes only effect would be that non-constant names might work better.

File: ecl/hqlcpp/hqlttcpp.cpp [hunk 2]
```diff
@@ -9365,6 +9369,7 @@ IHqlExpression * HqlLinkedChildRowTransformer::createTransformedBody(IHqlExpress
         {
             return QuickHqlTransformer::createTransformedBody(expr);
         }
+        break;
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlttcpp.cpp]: no, because the following code has if (false) followed by a break.

**Comment by ghalliday** [comment]
@richardkchapman see replies

**Pull request merged**